### ISSUE TITLE
[DOCUMENTATION] Coding style guidelines for contributions

### ISSUE THREAD

**Issue by Eversmile12** [alchemyplatform/create-web3-dapp#44]
It would be great to have fixed guidelines for common patterns and syntax. 

1. Don't think we should create new guidelines, we can borrow the ones from Google or AirBnB.
2. Linter needs to be setup to follow the guidelines
3. It would be great to have linter on PR request/merge -> [Issue about linter and testing on PRs](https://github.com/alchemyplatform/create-web3-dapp/issues/43)

Think it might be a good first issue.

**Comment by butovsky** [comment]
I propose the [Airbnb](https://github.com/airbnb/javascript) one: widespread, plus 95% of the rules are justified imo, and they are not so hard to follow in the end.

These guidelines are for JS, but, even if there are not any for TS from Airbnb, I guess the [plugin](https://www.npmjs.com/package/eslint-config-airbnb-typescript) will do the job most of the time (for a separate issue)

It's easy to setup with ESlint and also has a powerful autolinter. Might be a great idea to implement [precommit hooks](https://github.com/typicode/husky) with possible `npm run lint:fix`, which won't allow you to commit if some rules aren't followed. I guess, this can work in sense of applying some style guidelines in practice lol (also for a separate issue)

**Comment by Lavishq** [comment]
> 

why don't we use [StandardJs](https://standardjs.com/) to follow KISS principle and it will be more easier that way... i wanted to give a suggestion

**Comment by ef3n9r98** [comment]
Hey @Eversmile12, has someone already written the docs for this? I know some devs who'd be interested in working on this. 

Thanks!
